aspnetcore/blazor/tutorials/movie-database-app/part-8.md

```diff
@@ -0,0 +1,272 @@
+
+```razor
+<Paginator State="@pagination" />
+```
+
+Run the app and navigate to the movies `Index` page. You can page through the movie items at two movies per page:
+
+![Movie list showing the second page of two items](~/blazor/tutorials/movie-database-app/part-8/_static/paging-movies.png)
+
+The component is *interactive*. The page doesn't reload for paging to occur. The paging is performed live over the SignalR connection between the browser and the server, where the paging operation is performed on the server with the rendered result sent back to the client for the browser to display.
+
+Change <xref:Microsoft.AspNetCore.Components.QuickGrid.PaginationState.ItemsPerPage%2A> to a more reasonable value, such as 10 items per page:
+
+```diff
+- PaginationState pagination = new PaginationState { ItemsPerPage = 2 };
++ PaginationState pagination = new PaginationState { ItemsPerPage = 10 };
+```
+
+## Sortable `QuickGrid`
+
+Open the `Index` component (`Components/Pages/Movies/Index.razor`).
+
+Add `Sortable="true"` (<xref:Microsoft.AspNetCore.Components.QuickGrid.ColumnBase%601.Sortable%2A>) to the title <xref:Microsoft.AspNetCore.Components.QuickGrid.PropertyColumn%602>:
+
+```diff
+- <PropertyColumn Property="movie => movie.Title" />
++ <PropertyColumn Property="movie => movie.Title" Sortable="true" />
+```
+
+You can sort the `QuickGrid` by movie title by selecting the **:::no-loc text="Title":::** column. The page doesn't reload for sorting to occur. The sorting is performed live over the SignalR connection, where the sorting operation is performed on the server with the rendered result sent back to the client:
+
+![Movie list sorted by the Title column](~/blazor/tutorials/movie-database-app/part-8/_static/sorted-movies.png)
+
+## Use C# code and interactivity to search by title
+
+In an earlier part of the tutorial series, the `Index` component was modified to allow the user to filter movies by title. This was accomplished by:
+
+* Adding an HTML form that issues a GET request to the server with the user's title search string as a query string field-value pair (for example, `?titleFilter=road+warrior` if the user searches for "`road warrior`"):
+
+  ```html
+  <form action="/movies">
+      <input type="search" name="titleFilter" />
+      <input type="submit" value="Search" />
+  </form>
+  ```
+
+* Adding code to the component that obtains the title search string from the query string and uses it to filter the database's records:
+
+  ```csharp
+  [SupplyParameterFromQuery]
+  public string? TitleFilter { get; set; }
+
+  protected override void OnParametersSet()
+  {
+      if (!string.IsNullOrEmpty(TitleFilter))
+      {
+          movies = DB.Movie.Where(
+              s => !string.IsNullOrEmpty(s.Title) ? 
+                  s.Title.Contains(TitleFilter) : false);
+      }
+      else
+      {
+          movies = DB.Movie;
+      }
+  }
```

Comment:
Here's a simpler approach where you pass `FilteredMovies` as the model to the `QuickGrid`:

```suggestion
    [SupplyParameterFromQuery]
    public string TitleFilter { get; set; }

    IQueryable<Movie> FilteredMovies => DbFactory.CreateDbContext().Movie
        .Where(movie => movie.Title!.Contains(TitleFilter ?? ""));
```